test/redbird/crypto_test.exs

```diff
@@ -0,0 +1,96 @@
+defmodule Redbird.CryptoTest do
+  use Redbird.ConnCase, async: true
+
+  alias Redbird.Crypto
+
+  describe "sign_key/3" do
+    setup do
+      on_exit(fn -> Application.delete_env(:redbird, :key_base) end)
+    end
+
+    test "signs the key with configured key base" do
+      Application.put_env(:redbird, :key_base, "abcdef12345")
+
+      conn = Plug.Test.conn(:get, "/")
+      key = "somerediskey"
+
+      actual = Crypto.sign_key(key, conn)
+
+      refute actual =~ key
+      assert {:ok, ^key} = Crypto.verify_key(actual, conn)
+    end
+
+    test "signs the key with the conn secret key base when configured key base is not set" do
+      conn = signed_conn()
+      key = "somerediskey"
+
+      actual = Crypto.sign_key(key, conn)
+
+      refute actual =~ key
+      assert {:ok, ^key} = Crypto.verify_key(actual, conn)
+    end
+
+    test "errors when configured key base is not set and conn does not have secret key base" do
+      conn = Plug.Test.conn(:get, "/")
+      key = "somerediskey"
+
+      actual = Crypto.sign_key(key, conn)
+
+      assert {:error, message} = actual
+      assert message =~ ~r/key base is required/i
```

Comment:
👍  nice tests here. I found them easy to follow.

Reply:
Thanks!